src/reg.rs

```diff
@@ -0,0 +1,95 @@
+use core::{cmp, convert, ops};
+
+pub trait RegVal : Copy + Clone +
```

Comment:
Is there are reason `RegVal` needs to be abbreviated?

Reply:
Fixed